langlib/langlib-test/src/test/resources/test-src/xmllib_test.bal

```diff
@@ -382,6 +397,53 @@ function testElementChildrenNS() {
     assert(toNoNs.toString(), "<to>Irshad</to><to>Irshad</to>");
 }
 
+function testXMLIteratorInvocation() {
```

Comment:
Shouldn't the following result in a compilation error?

```ballerina
public function main() {
    'xml:Comment x = xml `<!-- foo -->`;
    var iterator = x.iterator();
}
```

Please check for the other non-text types too. 

This seems to fail on slp8 too, let's create an issue.

Reply:
#27848